js/script.js

```diff
@@ -526,10 +540,35 @@ const addTruncateSummary = () => {
           previousCard.classList.add('h-100');
         }
       }
-    })
-  };
+    });
+  }
 };
 
+function episodeLog(show) {
+  let userDetails = JSON.parse(localStorage.getItem('users'));
+  let episodeDropDownItem = document.getElementsByClassName('episode-tracker');
+
+  for (let i = 0; i < episodeDropDownItem.length; i++) {
+    let currentDropDownItem = episodeDropDownItem[i];
+    // eslint-disable-next-line no-loop-func
+    currentDropDownItem.addEventListener('click', (e) => {
+      e.preventDefault();
+      const episodeId = e.target.id;
+      let users = userDetails.map((user) => {
+        if (user.signedIn === true) {
+          let trackedShow = {
+            showId: show.id,
+            favEpisodes: episodeId,
+          };
+          user.episodeTracker.push(trackedShow);
+        }
+        return user;
+      });
+      localStorage.setItem('users', JSON.stringify(users));
```

Comment:
Good job stringifying and parsing the `users` when reading and writing from `localStorage`